### PR TITLE
mutt: Simplify `ghu` to a single file

### PR DESCRIPTION
**Pull request by gfontenot**
This was throwing errors, and was more complicated/flexible than I
really need it to be. So I simplified it a bit. It _seems_ to work now.

### CODE REVIEW

**Comment by gfontenot** [comment]
ping @gabebw @pbrisbin

File: tag-mutt/source/plugins/ghu [hunk 1]
```diff
@@ -28,9 +26,13 @@ get() {
 }
 
 parse() {
-  local url_re='https?://[^[:space:]>]+'
+  sed "/^List-Unsubscribe: /,/^\([A-Z].*: \| *$\)/!d" |\
+    sed -n 2p |\
+    sed "s/<\(.*\)>/\1/g" |\
+    xargs printf -- "%s " |\
+    sed "s/  *$//g"
```
Comment (gabebw) [inline tag-mutt/source/plugins/ghu]: Can `  *` be ` +`?

File: tag-mutt/source/plugins/ghu [hunk 2]
```diff
@@ -28,9 +26,13 @@ get() {
 }
 
 parse() {
-  local url_re='https?://[^[:space:]>]+'
```
Comment (gabebw) [inline tag-mutt/source/plugins/ghu]: Why did this disappear?

Comment (gfontenot) [inline tag-mutt/source/plugins/ghu]: Wasn't needed. This was used by Pat's version to further parse the result of the `sed` chain below.

File: tag-mutt/source/plugins/ghu (hunk 1 above)
Comment (gfontenot) [inline tag-mutt/source/plugins/ghu]: probably

File: tag-mutt/source/plugins/ghu [hunk 3]
```diff
@@ -28,9 +26,13 @@ get() {
...
+  sed "/^List-Unsubscribe: /,/^\([A-Z].*: \| *$\)/!d" |\
+    sed -n 2p |\
+    sed "s/<\(.*\)>/\1/g" |\
```
Comment (gabebw) [inline tag-mutt/source/plugins/ghu]: Should this be:

```
sed "s/^List-Unsubscribe: \(.*\)/\1/g"
```

instead of:

```
sed "s/<\(.*\)>/\1/g"```

Comment (gabebw) [inline tag-mutt/source/plugins/ghu]: It looks like `^List-Unsubscribe: \(.*\)` became `<\(.*\)>`.

**Comment by gabebw** [comment]
Disclaimer: I have never used mutt.

File: tag-mutt/source/plugins/ghu (hunk 2 above)
Comment (pbrisbin) [inline tag-mutt/source/plugins/ghu]: It looks like the `s/<\(.*\)\>/\1/` is playing this role now.

File: tag-mutt/source/plugins/ghu (hunk 3 above)
Comment (pbrisbin) [inline tag-mutt/source/plugins/ghu]: The text, I think, comes out like this:

```
List-Unsubscribe: <foo@bar.com>
    <http://the-url-you-want.com>
```

You're version would probably get the email.

**Comment by pbrisbin** [comment]
This LGTM to me. I think the trouble here was mostly GNU-vs-BSD and so I don't think it can be upstreamed (it's all BSD-ey now). But if this works for you, great!

File: tag-mutt/source/plugins/ghu (hunk 3 above)
Comment (gfontenot) [inline tag-mutt/source/plugins/ghu]: Exactly right. The actual URL gets dropped to the second line, so `sed -n -2p` gets that line, then this one parses the url out of the brackets.

**Comment by gfontenot** [comment]
@pbrisbin Yeah, I _started_ by switching things to the GNU versions, but it felt weird introducing `coreutils` as a dependency for this script. Once I started tweaking I assumed it would just live locally with me. BUT, I don't need `$BROWSER` or that wacky symlink from `usr/local/bin/sed` to `/usr/bin/sed` anymore!

**Pull request merged**